cmake/Utility.cmake

```diff
@@ -283,4 +283,180 @@ function(create_pipeline_tests)
 
 endfunction()
 
-#set_tests_properties(dependsTest12 PROPERTIES DEPENDS "baseTest1;baseTest2")
+#-------------------------------------------------------------------------------
+# This function generates a file ONLY if the MD5 between the "to be" generated file
+# and the current file are different. This will help reduce recompiles based on
+# the generation of files that are really the same.
+#-------------------------------------------------------------------------------
+function(cmpConfigureFileWithMD5Check)
+    set(options)
+    set(oneValueArgs CONFIGURED_TEMPLATE_PATH GENERATED_FILE_PATH )
+    cmake_parse_arguments(GVS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+ #   message(STATUS "   GVS_CONFIGURED_TEMPLATE_PATH: ${GVS_CONFIGURED_TEMPLATE_PATH}")
+ #   message(STATUS "   GVS_GENERATED_FILE_PATH: ${GVS_GENERATED_FILE_PATH}")
+
+    # Only Generate a file if it is different than what is already there.
+    if(EXISTS ${GVS_GENERATED_FILE_PATH} )
+        file(MD5 ${GVS_GENERATED_FILE_PATH} VERSION_HDR_MD5)
+        configure_file(${GVS_CONFIGURED_TEMPLATE_PATH}   ${GVS_GENERATED_FILE_PATH}_tmp  )
+
+        file(MD5 ${GVS_GENERATED_FILE_PATH}_tmp VERSION_GEN_HDR_MD5)
+        #message(STATUS "  File Exists, doing MD5 Comparison")
+
+        # Compare the MD5 checksums. If they are different then configure the file into the proper location
+        if(NOT "${VERSION_HDR_MD5}" STREQUAL "${VERSION_GEN_HDR_MD5}")
+            #message(STATUS "   ${VERSION_GEN_HDR_MD5}")
+            #message(STATUS "   ${VERSION_HDR_MD5}")
+            #message(STATUS "  Files differ: Replacing with newly generated file")
+            configure_file(${GVS_CONFIGURED_TEMPLATE_PATH}  ${GVS_GENERATED_FILE_PATH} )
+        else()
+            #message(STATUS "  NO Difference in Files")
+        endif()
+        file(REMOVE ${GVS_GENERATED_FILE_PATH}_tmp)
+    else()
+      # message(STATUS "  File does NOT Exist, Generating one...")
+      configure_file(${GVS_CONFIGURED_TEMPLATE_PATH} ${GVS_GENERATED_FILE_PATH} )
+    endif()
+
+endfunction()
+
+
+
+#-------------------------------------------------------------------------------
+# We are going to use Git functionality to create a version number for our package
+# The MAJOR.MINOR.PATCH is based off of YYYY.MM.DD
+# The TWEAK is the git hash of project.
+#-------------------------------------------------------------------------------
+function(cmpBuildDateRevisionString)
+  set(options)
+  set(oneValueArgs GENERATED_HEADER_FILE_PATH GENERATED_SOURCE_FILE_PATH
+                   NAMESPACE PROJECT_NAME EXPORT_MACRO VERSION_MACRO_PATH STRING_CLASS STRING_INCLUDE)
+  cmake_parse_arguments(GVS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+  if(NOT DEFINED GVS_STRING_CLASS)
+    set(GVS_STRING_CLASS "QString")
+  endif()
+
+  if(NOT DEFINED GVS_STRING_INCLUDE)
+    set(GVS_STRING_INCLUDE "<QtCore/QString>")
+  endif()
```

Comment:
Unless we want to change the license for NXRunner we should probably switch to standard library as the default